src/node_buffer.cc

```diff
@@ -1182,8 +1207,12 @@ void Swap16(const FunctionCallbackInfo<Value>& args) {
   THROW_AND_RETURN_UNLESS_BUFFER(env, args.This());
   SPREAD_ARG(args.This(), ts_obj);
 
-  for (size_t i = 0; i < ts_obj_length; i += 2) {
-    SWAP_BYTES(ts_obj_data, i, i + 1);
+  CHECK_EQ(ts_obj_length % 2, 0);
+  uint16_t* data16 = reinterpret_cast<uint16_t*>(ts_obj_data);
```

Comment:
You'll need to check:
```cpp
if (reinterpret_cast<uintptr_t>(ts_obj_data) % sizeof(uint16_t) == 0)
```
I'm pretty sure you'll need to make sure the pointer is aligned for the operation. Otherwise you can't cast to a `uint16_t`. Same with the other locations. I believe if this is false then you'll need to fall back to the slow mechanism of swapping bytes. /cc @bnoordhuis Can you confirm this?